docs/painless/painless-operators-boolean.asciidoc

```diff
@@ -0,0 +1,516 @@
+[[painless-operators-boolean]]
+=== Operators: Boolean
+
+[[boolean-not-operator]]
+==== Boolean Not
+
+Boolean not will flip a boolean value from true to false or false to true using the bang operator.  The format is a bang operator followed by an expression.
+
+*Grammar:*
+[source,ANTLR4]
+----
+boolean_not: '!' expression;
+----
+
+Note that def types will be assumed to be of the boolean type.  Any def type evaluated at run-time that does not represent a boolean will result in an error.  Non-boolean expressions will result in an error.
+
+*Examples:*
+[source,Java]
+----
+boolean x = !false; // declares the boolean variable x and sets it to the opposite of the false value
+boolean y = !x;     // declares the boolean variable y and sets it to the opposite of the boolean variable x
+def z = !y;         // declares the def variable z and sets it to the opposite of the boolean variable y
+----
+
+[[greater-than-operator]]
+==== Greater Than
+
+Greater than compares two numerical expressions where a resultant boolean value will be true if the left-side expression is a larger value than the right-side expression otherwise false.  The format is an expression, followed by the right angle operator, and a closing expression.
+
+*Grammar:*
+[source,ANTLR4]
+----
+greater_than: expression '>' expression;
+----
+
+A numeric promotion may occur during a greater than operation.  A def type evaluated at run-time will follow the same promotion table at run-time following whatever type def represents. Non-numeric expressions will result in an error.
+
```

Comment:
Maybe, "Using the greater than operator with non-numeric expressions results in an error."?

Reply:
Changed.